### ISSUE TITLE
Refactor BigInteger to BigInt for Dart2

### ISSUE THREAD

**Issue by RdeWilde** [stevenroose/dart-cryptoutils#3]
A new class has been added in Dart2 for 64+ bits integers. It should replace the BigInteger packages.

https://github.com/dart-lang/sdk/blob/master/CHANGELOG.md#dart-vm

**Comment by stevenroose** [comment]
I can't find the BigInt class in typed_data yet in the dart-lang/sdk repo.
Changing that will be a ton of work in PointyCastle, I guess. Dartcoin as
well uses BigInteger (from bignum package) for all it's crypto.

**Comment by RdeWilde** [comment]
https://github.com/dart-lang/sdk/blob/master/runtime/lib/bigint.dart

Yeah, I tried refactoring a big part of it, but BigInt is "lacking" a few functions.
```
- fromBytes(i, b)
- testBit(n)
- toByteArray()
- BigInt(bytes)
- copyTo(t)*
```

So I dunno what is best way to continue. Making util/helper functions to do these jobs for BigInt? Modify BigInteger package to play well with BigInt? Or maybe other alternatives.

Thing is Dart2 has limited ints to 64 bits. So for example Dartcoin doesn't work on Dart2 atm.

**Comment by stevenroose** [comment]
Dartcoin was intended to work in JS as well, so it assumed 53-bit ints. So nothing should really break in Dartcoin.

As for the BigInt methods, yeah it makes sense to have those methods in a [utils](https://github.com/dartcoin/dart-bitcoin/blob/master/lib/src/utils.dart) method. Maybe also think about if they can be avoided.

**Comment by RdeWilde** [comment]
For Dartcoin I created a seperate issue:
https://github.com/dartcoin/dart-bitcoin/issues/16

Still I think it could be a consideration to refactor to BigInt if it has same functionality, I'll just leave the issue open.

**Comment by techyrajeev** [comment]
compiler message: file:///private/var/www/flutter/.pub-cache/hosted/pub.dartlang.org/bignum-0.1.0/lib/src/big_integer_v8.dart:910:46: Error: The parameter 'm' of the method 'BigIntegerV8::modPowInt' has type bignum.v8::BigIntegerV8, which does not match the corresponding type in the overridden method (dynamic).
compiler message: Change to a supertype of dynamic (or, for a covariant parameter, a subtype).
compiler message:   BigIntegerV8 modPowInt(int e, BigIntegerV8 m) {
compiler message:                                              ^
compiler message: file:///private/var/www/flutter/.pub-cache/hosted/pub.dartlang.org/bignum-0.1.0/lib/bignum.dart:172:15: Context: This is the overriden method ('modPowInt').
compiler message:    BigInteger modPowInt(int e,  m);
compiler message:               ^
compiler message: file:///private/var/www/flutter/.pub-cache/hosted/pub.dartlang.org/bignum-0.1.0/lib/src/big_integer_v8.dart:456:28: Error: The parameter 'r' of the method 'BigIntegerV8::copyTo' has type bignum.v8::BigIntegerV8, which does not match the corresponding type in the overridden method (dynamic).
compiler message: Change to a supertype of dynamic (or, for a covariant parameter, a subtype).
compiler message:   void copyTo(BigIntegerV8 r) {
compiler message:                            ^

I am getting this error when using this packages

**Comment by stevenroose** [comment]
Solved by https://github.com/stevenroose/dart-cryptoutils/pull/5 and published as v0.4.0.
